src/main/scala/reactive/tweets/outgoing/TweetFlow.scala

```diff
@@ -0,0 +1,21 @@
+package reactive.tweets.outgoing
+
+import akka.actor.ActorRef
+import akka.http.scaladsl.model.ws.Message
+import akka.stream.scaladsl._
+import reactive.tweets.domain.Tweet
+import reactive.tweets.marshalling.TweetJsonProtocol
+
+trait TweetFlow extends TweetJsonProtocol {
+  private val tweetSource: Source[Tweet, ActorRef] = Source.actorPublisher[Tweet](TweetPublisher.props)
+  type TweetFilter = Tweet => Boolean
+  
+  def tweetFlow(tweetFilter: TweetFilter): Flow[Message, Message, Unit] =
+    Flow.wrap(Sink.ignore, tweetSource filter tweetFilter map toMessage)(Keep.left)
+
+  def tweetFlowOfUser(userName: String) = tweetFlow(_.user.name == userName)
+
+  def tweetFlowOfAll = tweetFlow(_ => true)
+
+  def tweetFlowOfHashTag(hashTag: String): Flow[Message, Message, Unit] = ???
```

Comment:
I heard you guys discussing hashTags when I already started reviewing, so this comment may be out-dated, but: should `???` be replaced with something real, or should a comment referring to the tutorial be added?